software/grapherrific.livecodescript

```diff
@@ -0,0 +1,1477 @@
+                     if char 1 of tTarget is "-" then
+                        -- negative, then append positive
+                        delete char 1 of tTarget
+                        put cr & tTarget & "," & tTurn after pMain[tDoc][tRecord][tTag]
+                     else
+                        -- positive, then append negative
+                        put cr & "-" & tTarget & "," & tTurn after pMain[tDoc][tRecord][tTag]
+                     end if
+                  end if
+               end repeat
+            end if
+         end repeat
+      end repeat
+      return tTemp["pointers"]
+   else
+      return "nothing to redo"
+   end if -- tRedoThisRec is not empy
+end redoTurn
+
+function listOfLinks pNode
+   put "function listOfLinks" into xX["write"];    devLog xX
+   -- pNode = record key of a node
+   -- returns a comma delimited list of the record keys of all the links attached to pNode
+   -- find the recs of the links attached to this node
+   put clearNegPairs(sMain[sNowDoc][pNode]["step"]) into tRecPtrs
+   put empty into tY -- will be list of link step records, if any links attached to this node
+   repeat for each line tPtr in tRecPtrs
+      repeat for each key tAkey in sMain[sNowDoc][item 1 of tPtr]
+         if tAkey is among the words of "bond next prev read skim trim bulk" then
+            repeat for each line tOtherPtr in sMain[sNowDoc][item 1 of tPtr][tAkey]
+               put item 1 of sMain[sNowDoc][item 1 of tOtherPtr]["myStep"] & "," after tY
+            end repeat
+         end if
+      end repeat
+   end repeat
+   return tY
+end listOfLinks
+
+command hideStep @pMain,pNums
+   put "command hideStep" into xX["write"];    devLog xX
+   put pNums["target"] into tHide
+   put pNums["turn"] into tTurn
+   put pNums["doc"] into tDoc
+
+   -- need clearNegPairs to work
+   
+   -- this works on nodes and links
+   -- but, call it on the links first, then on the node
+   -- negate these groups of pointers
+   -- stepA<tPtrA>myStepA
+   --                        linkA<tPtrAB>linkB
+   --                                               myStepB<tPtrB>stepB
+   put "prev next skim read bulk trim" into tLinkTags
+   put clearNegPairs(pMain[tDoc][tHide]["step"]) into tStepA
+   repeat for each line tPtrA in tStepA
+      put item 1 of tPtrA into tPtrA
+      put cr & "-" & tPtrA & "," & tTurn after pMain[tDoc][tHide]["step"]
+      put cr & "-" & tHide & "," & tTurn after pMain[tDoc][tPtrA]["myStep"]
+      repeat for each key tLinkA in pMain[tDoc][tPtrA]
+         if tLinkA is among the words of tLinkTags then
+            put clearNegPairs(pMain[tDoc][tPtrA][tLinkA]) into tLink
+            repeat for each line tPtrAB in tLink
+               put item 1 of tPtrAB into tPtrAB
+               put cr & "-" & tPtrAB & "," & tTurn after pMain[tDoc][tPtrA][tLinkA]
+               --               if tLinkA is "bond" then put "bond" into tLinkAmirror
+               if tLinkA is "next" then put "prev" into tLinkAmirror
```

Comment:
@matthewmaier It looks like this could be moved out of this repeat loop. Also I would use the switch control structure here for clarity.